modules/Bio/EnsEMBL/Variation/VariationFeature.pm

```diff
@@ -1968,7 +1967,118 @@ sub hgvs_genomic {
+Returntype  : Hash reference
+Exceptions  : Throws exception if VariationFeature can not be described relative to a Slice;
+              Throws exception if input is provided and value is not 0 or 1
+Caller      : general
+Status      : Experimental
+
+=cut
+
+sub spdi_genomic{
+
+  my $ref_feature = shift;
+  my $include_ref_allele = shift;
+
+  my %spdi;
+  if(!$include_ref_allele){ $include_ref_allele = 0; }
+  # If input is provided, it must have value '1' (include reference allele) or '0' (not include reference allele)
+  throw("Include reference allele must be a numeric value '1' or '0'.") unless ($include_ref_allele =~ m/^[01]$/);
+
+  my $ref_slice = $ref_feature->slice;
+  throw("This variation feature is not placed on a slice.") unless ($ref_slice->isa('Bio::EnsEMBL::Slice'));
+
+  my ($vf_start, $vf_end) = ($ref_feature->start, $ref_feature->end);
+
+  return {} if ($vf_start < 1 || $vf_end < 1);
+
+  my $vf_strand = $ref_feature->strand();
 
+  # set up sequence reference
+  my $syn = $ref_slice->get_all_synonyms('RefSeq_genomic');
+  my $reference_name = (defined $syn->[0] ? $syn->[0]->name() : $ref_feature->seq_region_name());
+
+  my @all_alleles = split(/\//,$ref_feature->allele_string());
+  my $ref_allele = shift @all_alleles;
+
+  my $spdi_ref_allele;
+  my $spdi_alt_allele;
+
+  # Create a spdi notation for each allele
+  foreach my $alt_allele (@all_alleles){
+
+    # Expand tandems before check for non nucleotide character
+    expand(\$alt_allele);
+
+    # Skip if the allele contains weird characters
+    next if $alt_allele =~ m/[^ACGT\-]/ig;
+
+    ##### vf strand is relative to slice
+    my $flip_allele = 0;
+    if( $vf_strand == -1 ){ $flip_allele = 1; }
+
+    my $chr_start = $ref_feature->start;
+    my $chr_end   = $ref_feature->end;
+
+    # If the variant is a substitution, deletion or indel
+    my $spdi_position = $chr_start - 1;
+
+    if($chr_start == $chr_end){
+      $spdi_ref_allele = $ref_allele;
+      # Variation is a deletion (1bp)
+      if($alt_allele eq '-'){ $spdi_alt_allele = ""; }
+      # Variation is a substitution
+      else{ $spdi_alt_allele = $alt_allele; }
+    }
+    # Variation is an insertion
+    elsif($ref_allele eq '-'){
+      $spdi_position = $chr_end;
+      $spdi_ref_allele = "";
+      $spdi_alt_allele = $alt_allele;
+    }
+    # Variation is a deletion (>1bp)
+    elsif($alt_allele eq '-'){
+      $spdi_ref_allele = $ref_allele;
+      my $ref_size = length($spdi_ref_allele);
+      # If reference allele (e.g. deleted sequence) size > 20bp then spdi notation for the deletion is NC_000002.12:47403326:29:
+      $spdi_ref_allele = $ref_size unless ( $ref_size <= 20 );
+      $spdi_alt_allele = "";
+    }
+    # Variation is an indel
+    else{
+      $spdi_ref_allele = $ref_allele;
+      $spdi_alt_allele = $alt_allele;
+    }
+
+    reverse_comp(\$spdi_ref_allele) if( $flip_allele == 1 && $spdi_ref_allele && $spdi_ref_allele =~ m/^[A-Z]+/i);
```

Comment:
Not sure why pattern match A-Z?

Reply:
Sometimes the ref_allele can be a number, for example a deletion of 21 bp (NC_000002.12:47404148:10:) in this case the reverse would be 12. So, I put the pattern to make sure it would only reverse sequences with letters.

Reply:
It might be useful to add a comment about this, as earlier only pattern match  $alt_allele =~ m/[^ACGT\-]/ig;
What is returned if need to reverse on longer deletions?
Should the reverse_comp of ref and alt be grouped in an if (flip_allele == 1)